psga/config/psga-schema-sars-cov-2.yaml

```diff
@@ -394,84 +394,115 @@ input_fields:
 output_fields:
   - field_name: STATUS
     field_type: string
+    ui:
```

Comment:
Is this defined UI because you foresee more for the UI coming, I would have thought `label` would be generic, although this doesn't fit well with multi lingual setups if you're now looking at PGX. It's potentially a separate mapping of the field slug (name in this case) to a language translation. Depending on how you forsee that working, it will probably be a separate config that links the field to the label. 

I see no issue for now to provide the feature, especially as multi lingual was descoped way back.

Reply:
This is as per design doc https://confluence.congenica.net/display/PSG/Metadata+Display. While we use name in the field_name attribute which is computer readable, generally in the data model we use the term slug. To me ui.name seems reasonably obvious that it's a name for a ui so a display name. 

While I wouldn't say i18n is descoped we're just not doing it now. 

For internationalisation I believe an appropriate approach for such fields would be to use a value of ${pipeline.sars-cov-2.field.STATUS} and then provide a lookup file by language with the pipeline. You'd lookup using pipeline.sars-cov-2.field.STATUS as the key.

At the moment we're not bothering with the lookup because we're not trying to do internationalisation at the moment. These are UI hints for the GUI and not for interpretation by the UI. Hence I would rather group then together under the ui and make it clear that the structure under UI needs to simply be passed to the UI for interpretation. This may be les true when we do internationalisation but that would simply be examining all attributes under UI and seeing if they are encapsulated in ${...} and if so do the lookup into the translation file.

Reply:
I hadn't seen the design doc. 

The separate translations file is what I had in mind too, it's possible that you could utilise the django translations mechanism to get and set translations. A discussion for another day though I'm sure.